src/mpid/ch4/shm/ipc/gpu/gpu_init.c

```diff
@@ -47,13 +47,16 @@ static void ipc_handle_free_hook(void *dptr)
         MPIR_Assert(mpl_err == MPL_SUCCESS);
 
         MPIR_GPU_query_pointer_attr(pbase, &gpu_attr);
-        local_dev_id = MPL_gpu_get_dev_id_from_attr(&gpu_attr);
-        global_dev_id = MPL_gpu_local_to_global_dev_id(local_dev_id);
-
-        for (int i = 0; i < MPIR_Process.local_size; ++i) {
-            MPL_gavl_tree_t track_tree = MPIDI_GPUI_global.ipc_handle_track_trees[i][global_dev_id];
-            mpl_err = MPL_gavl_tree_delete_range(track_tree, pbase, len);
-            MPIR_Assert(mpl_err == MPL_SUCCESS);
+        if (gpu_attr.type == MPL_GPU_POINTER_DEV) {
```

Comment:
don't do it here. Check the device type in the `zeMemFree` in MPL before call the hooks.

Reply:
It would add an extra pointer query in the path and prevent creating hooks which distinguish between different memory types (i.e. USM host and USM shared) in the future if we moved the check to the the `zeMemFree` call